### PR TITLE
Add `#id`-based navigation.

### PR DESCRIPTION
**Pull request by bnjmnt4n**

### CODE REVIEW

**Comment by bnjmnt4n** [comment]
@joyceerhl have updated it, should be working fine now. I think it depended on some of the changes in #41, which are inside now that I’ve rebased on master.

**Comment by bnjmnt4n** [comment]
I’ve updated to use kebab-cased mentor names as the IDs instead. A quick check shows that all mentor names are unique (currently).

```
> var d = require('./src/mentors')
> d.mentors.length
215
> [...new Set(d.mentors.map(({ name }) => name))].length
215
> [...new Set(d.mentors.map(({ name }) => name.toLowerCase().replace(' ', '-').replace('(', '').replace(')', '')))].length
215
```

File: src/App.js [hunk 1]
```diff
@@ -8,29 +8,35 @@ import { mentors } from "./mentors.json";
...
       let hash = window.location.hash.slice(1);
-      let num = hash.length ? Number(hash) : "";
+      let mentorName = hash.length ? String(hash) : "";
+      let mentorId = mentorName.length ? mentorNames.indexOf(mentorName) : -1;
 
-      if (typeof num == "number" && !isNaN(num) && mentors[num]) {
-        if (!isModalOpen || num !== activeMentorId) {
-          activateModal(num);
+      if (mentorId !== -1) {
+        if (!isModalOpen || mentorId !== activeMentorId) {
```
Comment (wei2912) [inline src/App.js:30]: Feels like this check should be done in `activateModal` instead, what do you think?

File: src/App.js [hunk 2]
```diff
@@ -8,29 +8,35 @@ import { mentors } from "./mentors.json";
 import "./App.css";
 
 const visibleMentorIds = mentors.map((mentor, index) => index);
+
+// Convert names into usable `#`-ids.
+const mentorNames = mentors.map(({ name }) =>
```
Comment (wei2912) [inline src/App.js:13]: Could I propose using `mentorHashes` instead? Just to keep it separate from the idea of the mentors' names.

File: src/App.js (hunk 1 above)
Comment (bnjmnt4n) [reply src/App.js:30]: I guess it could, although these cases would only ever occur here. I have no strong opinions: feel free to add it if you want.

Comment (wei2912) [reply src/App.js:30]: Hmm alright, I guess it's not a big issue.

**Pull request merged**